apps/payments/next/providers/AppLocalizationProvider.tsx

```diff
@@ -0,0 +1,56 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+'use client';
+
+import React, { createContext, useContext, useState, useEffect } from 'react';
+import { ReactLocalization } from '@fluent/react';
+import { useParams } from 'next/navigation';
+import { getLocaleFromRequest } from '@fxa/shared/l10n';
+import {
+  LocalizerClient,
+  LocalizerBindingsClient,
+} from '@fxa/shared/l10n/client';
+
+const AppLocalizationContext = createContext<ReactLocalization | undefined>(
+  undefined
+);
+
+export function AppLocalizationProvider({
+  children,
+}: {
+  children: React.ReactNode;
+}) {
+  const [l10n, setL10n] = useState<ReactLocalization>();
+  const [isLocalized, setIsLocalized] = useState<boolean>(true);
+
+  const params = useParams();
+  const locale = getLocaleFromRequest(params, null);
+
+  useEffect(() => {
+    const setLocalization = async () => {
+      const bindings = new LocalizerBindingsClient();
+      bindings.fetchResource(
+        `${bindings.opts.translations.basePath}/${locale}/main.ftl`
+      );
+      const localizerClient = new LocalizerClient(bindings);
+      const l10nPromise = await localizerClient.setupReactLocalization(locale);
+      await Promise.resolve(l10nPromise).then((res) => setL10n(res.l10n));
```

Comment:
This is not necessary, the promise is already resolved on line 38. 

This line, line 39, could be replaced with the following, or how it's currently done in the layout.tsx implementation.
```
setL10n(l10nPromise.l10n)
```